streaming.md

```diff
@@ -91,10 +91,12 @@ In particular, the size of the `expectedContentTypes` field (as well as the `out
 
 ### InputFrame
 `InputFrames` represent incoming data to the function. Upon reception of an `InputFrame`, an invoker
-1. SHOULD deserialize the *payload* according to the *contentType* field, which MUST be set and represent a concrete MIME type (*i.e.* not a wildcard type). The data structure that corresponds to the result of de-serialization is runtime-specific and is beyond the scope of this specification. To the best extent though, an invoker SHOULD use data structures that are *idiomatic* to the target runtime and pass them to the function if feasible. As an example, if the incoming `contentType` is `application/json` and `payload` contains the bytes of `{"foo": 5}`, then the invoker may use a *map* or a *dictionary* with key `foo`.
-2. MAY use any additional hints carried in the custom `headers` field to decide how to deserialize the incoming data
-3. MAY decide to craft a data structure that embeds the `contentType` field, as well as some or all of the custom `headers` and pass that to the function
-4. MUST forward the result of the deserialization to the i-th input *stream* of the function, where `i` is the value of the `argIndex` field.
+1. MUST deserialize the *payload* according to the *contentType* field, which SHOULD be set and represent a concrete MIME type (*i.e.* not a wildcard type) or default to `application/octet-stream`. The data structure that corresponds to the result of de-serialization is runtime-specific and is beyond the scope of this specification. To the best extent though, an invoker SHOULD use data structures that are *idiomatic* to the target runtime and pass them to the function if feasible. As an example, if the incoming `contentType` is `application/json` and `payload` contains the bytes of `{"foo": 5}`, then the invoker may use a *map* or a *dictionary* with key `foo`.
+2. If the *contentType* is not supported, an invoker MUST craft a gRPC status error of code [InvalidArgument](https://github.com/grpc/grpc/blob/2a7191abe717d3ce33b5fcbfd6bd118367499cde/doc/statuscodes.md#status-codes-and-their-use-in-grpc) and the error message MUST start with `"Invoker: Unsupported Media Type"`. An invoker MAY append error details to that message prefix. These details SHOULD start with punctuation to separate them from the error message prefix.
+3. MUST fail this rpc invocation if it cannot otherwise successfully deserialize data using such a MIME type
```

Comment:
Should a particular gRPC status code be used?

Reply:
Maybe UNKNOWN? WDYT @ericbottard?

Reply:
done